In [10]:
var = 2

In [11]:
def ifmodifier(test):
    return 'true, man!' if test else 'false, man!'


In [12]:
ifmodifier(True)

'true, man!'

In [13]:
ifmodifier(False)

'false, man!'

In [15]:
import os
os.listdir('.')

['supermarkets',
 '.ipynb_checkpoints',
 'supermarkets.zip',
 'Untitled.ipynb',
 'text_generator.py']

In [16]:
import pandas

In [17]:
os.chdir('supermarkets')

In [19]:
df1 =pandas.read_csv('supermarkets.csv')

In [20]:
df1

ID          Address            City             State Country  \
0   1     3666 21st St   San Francisco          CA 94114     USA   
1   2   735 Dolores St   San Francisco          CA 94119     USA   
2   3      332 Hill St   San Francisco  California 94114     USA   
3   4     3995 23rd St   San Francisco          CA 94114     USA   
4   5  1056 Sanchez St   San Francisco        California     USA   
5   6  551 Alvarado St   San Francisco          CA 94114     USA   

           Name  Employees  
0       Madeira          8  
1   Bready Shop         15  
2   Super River         25  
3    Ben's Shop         10  
4       Sanchez         12  
5    Richvalley         20

In [21]:
df1.shape

(6, 7)

In [22]:
df1.set_index("ID")

Address            City             State Country          Name  \
ID                                                                            
1      3666 21st St   San Francisco          CA 94114     USA       Madeira   
2    735 Dolores St   San Francisco          CA 94119     USA   Bready Shop   
3       332 Hill St   San Francisco  California 94114     USA   Super River   
4      3995 23rd St   San Francisco          CA 94114     USA    Ben's Shop   
5   1056 Sanchez St   San Francisco        California     USA       Sanchez   
6   551 Alvarado St   San Francisco          CA 94114     USA    Richvalley   

    Employees  
ID             
1           8  
2          15  
3          25  
4          10  
5          12  
6          20

In [23]:
df2 = pandas.read_json("supermarkets.json")

In [28]:
df2.set_index("ID")

Address           City Country  Employees         Name  \
ID                                                                   
1      3666 21st St  San Francisco     USA          8      Madeira   
2    735 Dolores St  San Francisco     USA         15  Bready Shop   
3       332 Hill St  San Francisco     USA         25  Super River   
4      3995 23rd St  San Francisco     USA         10   Ben's Shop   
5   1056 Sanchez St  San Francisco     USA         12      Sanchez   
6   551 Alvarado St  San Francisco     USA         20   Richvalley   

               State  
ID                    
1           CA 94114  
2           CA 94119  
3   California 94114  
4           CA 94114  
5         California  
6           CA 94114

In [33]:
df3 = pandas.read_excel('supermarkets.xlsx', sheetname="Ark2")

In [40]:
df4 = pandas.read_csv('supermarkets.csv', sep=';')
df4

ID,Address,City,State,Country,Name,Employees
0  1, 3666 21st St,San Francisco, CA 94114, USA, ...
1  2, 735 Dolores St,San Francisco, CA 94119, USA...
2  3, 332 Hill St, San Francisco,California 94114...
3  4, 3995 23rd St, San Francisco, CA 94114,USA, ...
4  5,1056 Sanchez St,San Francisco,California,USA...
5  6,551 Alvarado St,San Francisco,CA 94114,USA,R...

In [41]:
dfjson =  df2
dfjson

Address           City Country  Employees  ID         Name  \
0     3666 21st St  San Francisco     USA          8   1      Madeira   
1   735 Dolores St  San Francisco     USA         15   2  Bready Shop   
2      332 Hill St  San Francisco     USA         25   3  Super River   
3     3995 23rd St  San Francisco     USA         10   4   Ben's Shop   
4  1056 Sanchez St  San Francisco     USA         12   5      Sanchez   
5  551 Alvarado St  San Francisco     USA         20   6   Richvalley   

              State  
0          CA 94114  
1          CA 94119  
2  California 94114  
3          CA 94114  
4        California  
5          CA 94114

In [42]:
dfjson.set_index("Address")

City Country  Employees  ID         Name  \
Address                                                              
3666 21st St     San Francisco     USA          8   1      Madeira   
735 Dolores St   San Francisco     USA         15   2  Bready Shop   
332 Hill St      San Francisco     USA         25   3  Super River   
3995 23rd St     San Francisco     USA         10   4   Ben's Shop   
1056 Sanchez St  San Francisco     USA         12   5      Sanchez   
551 Alvarado St  San Francisco     USA         20   6   Richvalley   

                            State  
Address                            
3666 21st St             CA 94114  
735 Dolores St           CA 94119  
332 Hill St      California 94114  
3995 23rd St             CA 94114  
1056 Sanchez St        California  
551 Alvarado St          CA 94114

In [43]:
dfjson

Address           City Country  Employees  ID         Name  \
0     3666 21st St  San Francisco     USA          8   1      Madeira   
1   735 Dolores St  San Francisco     USA         15   2  Bready Shop   
2      332 Hill St  San Francisco     USA         25   3  Super River   
3     3995 23rd St  San Francisco     USA         10   4   Ben's Shop   
4  1056 Sanchez St  San Francisco     USA         12   5      Sanchez   
5  551 Alvarado St  San Francisco     USA         20   6   Richvalley   

              State  
0          CA 94114  
1          CA 94119  
2  California 94114  
3          CA 94114  
4        California  
5          CA 94114

In [44]:
dfjson = dfjson.set_index("Address")

In [45]:
dfjson

City Country  Employees  ID         Name  \
Address                                                              
3666 21st St     San Francisco     USA          8   1      Madeira   
735 Dolores St   San Francisco     USA         15   2  Bready Shop   
332 Hill St      San Francisco     USA         25   3  Super River   
3995 23rd St     San Francisco     USA         10   4   Ben's Shop   
1056 Sanchez St  San Francisco     USA         12   5      Sanchez   
551 Alvarado St  San Francisco     USA         20   6   Richvalley   

                            State  
Address                            
3666 21st St             CA 94114  
735 Dolores St           CA 94119  
332 Hill St      California 94114  
3995 23rd St             CA 94114  
1056 Sanchez St        California  
551 Alvarado St          CA 94114

In [57]:
dfjson.loc["3666 21st St"::2,"City"]

Address
3666 21st St       San Francisco
332 Hill St        San Francisco
1056 Sanchez St    San Francisco
Name: City, dtype: object

In [69]:
dfjson.loc[:, ['City', 'Country']]

City Country
Address                               
3666 21st St     San Francisco     USA
735 Dolores St   San Francisco     USA
332 Hill St      San Francisco     USA
3995 23rd St     San Francisco     USA
1056 Sanchez St  San Francisco     USA
551 Alvarado St  San Francisco     USA

In [72]:
dfjson.loc[:, ['City', 'Country']]

City Country
Address                               
3666 21st St     San Francisco     USA
735 Dolores St   San Francisco     USA
332 Hill St      San Francisco     USA
3995 23rd St     San Francisco     USA
1056 Sanchez St  San Francisco     USA
551 Alvarado St  San Francisco     USA

`loc[rows,cols]`

rows/cols can be:
- list with labels (e.g., loc[['735 Dolores St', '551 Alvarado St'],['City', 'ID', 'Name']])
- positional list of booleans (e.g., [[True, False, True, True],[True, False,True, False, False, True]])
- named slices (e.g., ['332 Hill St': , 'City':'Employess':-1])
- an one-argument callback that returns a valid pandas indexing(DataFrame, Series or Panel)
- combinations of those formats(i.e., use a format for rows different than o use for cols and vice versa)

In [85]:
s= dfjson['Country']

In [100]:
from geopy.geocoders import Nominatim

In [101]:
dfjson

City Country  Employees  ID         Name  \
Address                                                              
3666 21st St     San Francisco     USA          8   1      Madeira   
735 Dolores St   San Francisco     USA         15   2  Bready Shop   
332 Hill St      San Francisco     USA         25   3  Super River   
3995 23rd St     San Francisco     USA         10   4   Ben's Shop   
1056 Sanchez St  San Francisco     USA         12   5      Sanchez   
551 Alvarado St  San Francisco     USA         20   6   Richvalley   

                            State  
Address                            
3666 21st St             CA 94114  
735 Dolores St           CA 94119  
332 Hill St      California 94114  
3995 23rd St             CA 94114  
1056 Sanchez St        California  
551 Alvarado St          CA 94114

In [116]:
nom = Nominatim()

In [130]:
df1['Full Address']=df1['Address'] + ', ' + df1['City'] + ', ' + df1['State'] + ', ' + df1["Country"]

In [131]:
address = df1.ix[2, 'Full Address']
print address
print type(address)
n = nom.geocode(address)

 332 Hill St,  San Francisco, California 94114, USA
<type 'str'>


In [149]:
df1['Coordinates']=df1['Full Address'].apply(nom.geocode)

AttributeError: 'Series' object has no attribute 'point'

In [155]:
df1['Coordinates'] = df1['Coordinates'].apply(lambda x: x.point if x else None)

In [161]:
df1['Coordinates'].first?

Object `first` not found.


In [163]:
a = df1['Coordinates']

Point(37.756488877551, -122.429343346939, 0.0)

In [169]:
type(df1['Coordinates'][0])

geopy.point.Point